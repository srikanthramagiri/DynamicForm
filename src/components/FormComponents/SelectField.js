import React, { useState } from "react";
import { Col, Form, Row } from "react-bootstrap";
import AlertError from "./AlertError";
import Label from "./Label";
import PropTypes from "prop-types";

export default function SelectField(props) {
  let { values, validation, label, name, changeHandler } = props;
  let { regex, errorMessage } = validation ? validation : {};
  let [isValidSelect, setIsValidSelect] = useState(true);
  let [country, setCountry] = useState();

  const onChangeHandler = (e) => {
    let reg = new RegExp(regex);
    if (e.target.value === "") {
      setIsValidSelect(false);
    } else {
      setIsValidSelect(true);
    }
    changeHandler(e);
    setCountry(e.target.value);
  };

  const onBlurHandler = (e) => {
    if (e.target.value === "") {
      setIsValidSelect(false);
    } else {
      setIsValidSelect(true);
    }
  };

  return (
    <Row>
      <Col lg={6} md={6}>
        <Label name={name} label={label} />
      </Col>
      <Col lg={6} md={6}>
        <Form.Control
          as={'select'}
          data-test='select-field'
          value={country}
          name={name}
          onChange={(e) => onChangeHandler(e)}
          onBlur={(e) => onBlurHandler(e)}
        >
          <option value="">Select</option>
          {values.map((option,i) => (
            <option value={option} key={i}>{option}</option>
          ))}
        </Form.Control>
      </Col>
      <Col lg={6} />
      <Col lg={6} md={6}>
        {!isValidSelect ? (
          <AlertError variant={"danger"} message={errorMessage} />
        ) : (
          ""
        )}
      </Col>
    </Row>
  );
}

SelectField.propTypes = {
  values: PropTypes.array,
  validation: PropTypes.object,
  label: PropTypes.string,
  name: PropTypes.string,
  changeHandler: PropTypes.func,
};
