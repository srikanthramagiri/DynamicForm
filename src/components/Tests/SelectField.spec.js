import {shallow} from 'enzyme'
import SelectField from '../FormComponents/SelectField'

describe('select field component', ()=> {
    const optionsValues=  ["India", "srilanka", "Algeria", "Andorra"]
    const changeHandlerSpy = jest.fn();
    const wrapper = shallow(<SelectField values={optionsValues} changeHandler={changeHandlerSpy}/>)
    const Select =  wrapper.find(`[data-test='select-field']`)
    it('should render correctly', ()=> {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).not.toBeNull()
    })
    it('should have one input text area',()=> {
        expect(Select).not.toBeNull()
    }) 
    it('change handler', ()=> {
        Select.simulate('change', {target: {value: 'india'}})
        expect(changeHandlerSpy).toHaveBeenCalled()
    })
    it('change handler when emty value', ()=> {
        Select.simulate('change', {target: {value: ''}})
        expect(changeHandlerSpy).toHaveBeenCalled()
    })
})