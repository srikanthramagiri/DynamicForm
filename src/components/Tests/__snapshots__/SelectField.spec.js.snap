// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select field component should render correctly 1`] = `ShallowWrapper {}`;
